packages/sdk_connectivity_cubit/lib/src/sdk_connectivity_cubit.dart

```diff
@@ -0,0 +1,89 @@
+import 'dart:async';
+
+import 'package:bip39/bip39.dart';
+import 'package:breez_sdk_liquid/breez_sdk_liquid.dart';
+import 'package:connectivity_plus/connectivity_plus.dart';
+import 'package:credentials_manager/credentials_manager.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_breez_liquid/flutter_breez_liquid.dart';
+import 'package:sdk_connectivity_cubit/sdk_connectivity_cubit.dart';
+
+class SdkConnectivityCubit extends Cubit<SdkConnectivityState> {
+  final CredentialsManager credentialsManager;
+  final BreezSDKLiquid liquidSDK;
+
+  SdkConnectivityCubit({
+    required this.credentialsManager,
+    required this.liquidSDK,
+  }) : super(SdkConnectivityState.disconnected);
+
+  Future<void> register() async {
+    final mnemonic = generateMnemonic(strength: 128);
+    await _connect(mnemonic, storeMnemonic: true);
+  }
+
+  Future<void> restore({required String mnemonic}) async {
+    await _connect(mnemonic, storeMnemonic: true);
+  }
+
+  Future<void> reconnect() async {
+    try {
+      final mnemonic = await credentialsManager.restoreMnemonic();
+      if (mnemonic != null) {
+        await _connect(mnemonic);
+      } else {
+        throw Exception("No mnemonics");
+      }
+    } catch (e) {
+      await _retryUntilConnected();
+    }
+  }
+
+  Future<void> _connect(String mnemonic, {bool storeMnemonic = false}) async {
+    try {
+      emit(SdkConnectivityState.connecting);
+
+      final config = await AppConfig.instance();
+      final req = ConnectRequest(mnemonic: mnemonic, config: config.sdkConfig);
+      await liquidSDK.connect(req: req);
+
+      _startSyncing();
+
+      if (storeMnemonic) {
+        await credentialsManager.storeMnemonic(mnemonic: mnemonic);
+      }
+
+      emit(SdkConnectivityState.connected);
+    } catch (e) {
+      if (storeMnemonic) {
+        await credentialsManager.deleteMnemonic();
```

Comment:
@roeierez We use deleteMnemonic here on failed register & restore attempts.

It should have been introduced as part of this pr instead of

https://github.com/breez/l-breez/pull/123